File: Lesson2/Lesson2/Controllers/ProductController.cs
using Microsoft.AspNetCore.Mvc;


namespace StudentManagement.Controllers
{
    public class ProductController : Controller
    {

        public ActionResult Detail(int? id)
        {
            if (id == null)
            {
                ViewBag.Message = "Error: Product ID is required";
            }
            else
            {
                ViewBag.Message = "Product ID = " + id;
            }

            return View();
        }

        public ActionResult Category(string name)
        {
            if (string.IsNullOrEmpty(name))
            {
                ViewBag.Message = "Error: Category name is required";
            }
            else
            {
                ViewBag.Message = "Category = " + name;
            }

            return View();
        }
    }
}

File: SpringBoot/studentmanagement/src/main/java/com/example/studentmanagement/controller/ProductController.java
package com.example.studentmanagement.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

@Controller
@RequestMapping("/product")
public class ProductController {

    @GetMapping("/detail/{id}")
    @ResponseBody
    public String detail(@PathVariable Integer id) {

        if (id <= 0) {
            return "Error: Product ID must be greater than 0";
        }

        return "Product ID = " + id;
    }

    @GetMapping("/category")
    @ResponseBody
    public String category(
            @RequestParam(required = false) String name) {

        if (name == null || name.trim().isEmpty()) {
            return "Error: Category name is required";
        }

        return "Category = " + name;
    }
}